Unit tests fail on Windows

Two of the unit tests fail on Windows because Windows doesn't like it if you try to delete file that is still open. (You can get away with that on *nix.)

I will submit a pull request shortly. Here are the errors:

Running TestSuite
Tests run: 333, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 5.816 sec <<< FAILURE! - in TestSuite
testDelete(io.atomix.catalyst.buffer.FileBufferTest)  Time elapsed: 0.018 sec  <<< FAILURE!
java.lang.RuntimeException: 
java.nio.file.FileSystemException: target\test-files\46cd32f5-9ced-4f36-b56c-2448732b90e4: The process cannot access the file because it is being used by another process.

	at io.atomix.catalyst.buffer.FileBufferTest.testDelete(FileBufferTest.java:72)
Caused by: java.nio.file.FileSystemException: 
target\test-files\46cd32f5-9ced-4f36-b56c-2448732b90e4: The process cannot access the file because it is being used by another process.

	at io.atomix.catalyst.buffer.FileBufferTest.testDelete(FileBufferTest.java:72)

testDelete(io.atomix.catalyst.buffer.MappedBufferTest)  Time elapsed: 0.003 sec  <<< FAILURE!
java.lang.RuntimeException: 
java.nio.file.FileSystemException: target\test-files\3eab3429-a82b-4b82-945b-3eccfaff4067: The process cannot access the file because it is being used by another process.

	at io.atomix.catalyst.buffer.MappedBufferTest.testDelete(MappedBufferTest.java:74)
Caused by: java.nio.file.FileSystemException: 
target\test-files\3eab3429-a82b-4b82-945b-3eccfaff4067: The process cannot access the file because it is being used by another process.

	at io.atomix.catalyst.buffer.MappedBufferTest.testDelete(MappedBufferTest.java:74)

--- a/buffer/src/main/java/io/atomix/catalyst/buffer/FileBytes.java
+++ b/buffer/src/main/java/io/atomix/catalyst/buffer/FileBytes.java
@@ -639,6 +639,7 @@ public class FileBytes extends AbstractBytes {
    */
   public void delete() {
     try {
+      close();
       Files.delete(file.toPath());
     } catch (IOException e) {
       throw new RuntimeException(e);

--- a/buffer/src/main/java/io/atomix/catalyst/buffer/MappedBytes.java
+++ b/buffer/src/main/java/io/atomix/catalyst/buffer/MappedBytes.java
@@ -112,11 +112,17 @@ public class MappedBytes extends NativeBytes {
     return this;
   }
 
+  @Override
+  public void close() {
+    ((MappedMemory) memory).close();
+  }
+  
   /**
    * Deletes the underlying file.
    */
   public void delete() {
     try {
+      close();
       Files.delete(file.toPath());
     } catch (IOException e) {
       throw new RuntimeException(e);

--- a/buffer/src/main/java/io/atomix/catalyst/buffer/util/MappedMemory.java
+++ b/buffer/src/main/java/io/atomix/catalyst/buffer/util/MappedMemory.java
@@ -84,4 +84,9 @@ public class MappedMemory extends NativeMemory {
     ((MappedMemoryAllocator) allocator).release();
   }
 
+  public void close() {
+    free();
+    ((MappedMemoryAllocator) allocator).close();
+  }
+  
 }

--- a/buffer/src/main/java/io/atomix/catalyst/buffer/util/MappedMemoryAllocator.java
+++ b/buffer/src/main/java/io/atomix/catalyst/buffer/util/MappedMemoryAllocator.java
@@ -107,16 +107,20 @@ public class MappedMemoryAllocator implements MemoryAllocator<MappedMemory> {
     return newMemory;
   }
 
+  public void close() {
+    try {
+      file.close();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+  
   /**
    * Releases a reference from the allocator.
    */
   void release() {
     if (referenceCount.decrementAndGet() == 0) {
-      try {
-        file.close();
-      } catch (IOException e) {
-        throw new RuntimeException(e);
-      }
+      close();
     }
   }
 

